services/wireguard/service/service_unix.go

```diff
@@ -65,39 +66,39 @@ type Manager struct {
 }
 
 // ProvideConfig provides the config for consumer
-func (manager *Manager) ProvideConfig(publicKey json.RawMessage, pingerPort func(int) int) (session.ServiceConfiguration, session.DestroyCallback, error) {
+func (manager *Manager) ProvideConfig(publicKey json.RawMessage, params *traversal.Params) (session.ServiceConfiguration, session.DestroyCallback, *traversal.Params, error) {
```

Comment:
That is how ProvideConfig is designed. Service interface accepts this method, but at the same time is used as a Manager and as a ConfigNegotiator. Sometimes we need to pass params further into ConfigNegotiator and sometimes return them earlier. 

Will be bundling that into struct as @vkuznecovas suggested, so it will not look that ugly :)